lib/devise/controllers/sign_in_out.rb

```diff
@@ -114,6 +115,12 @@ def expire_data_after_sign_in!
       end
 
       alias :expire_data_after_sign_out! :expire_data_after_sign_in!
+
+      def update_tracked_fields?(resource, warden)
+        resource.respond_to?(:update_tracked_fields!) &&
+          !warden.env['devise.skip_trackable'] &&
+          (!resource.respond_to?(:active_for_authentication?) || resource.active_for_authentication?)
```

Comment:
The older hook was not checking this conditional, was it?
I thought that the `activatable` [hook](https://github.com/plataformatec/devise/blob/master/lib/devise/hooks/activatable.rb#L7) would solve this already.

Reply:
We need this check otherwise this test will fail:

https://github.com/plataformatec/devise/blob/76525939f659232b15f99ae356fb6262ae9bab2b/test/integration/trackable_test.rb#L72

I'm not sure, but I think it was being checked by `warden.authenticated?(options[:scope])` on the trackable hook.

Reply:
Yep, I have to look at it again, but I added it because some test was failing. But if @tegon says that this was already checked by `activatable` hook, maybe the problem is related with something not being stubbed in the test. I'll try to retake it as soon as I find the time, but if you or anybody can do it now please go ahead.

Reply:
Could you give me permission to commit in this branch? I can continue the work that you've done here if you don't mind :)

Reply:
Done @feliperenan ! I have added you as a collaborator to the fork, so you now have push access to this branch. Thanks! :)

Reply:
> I'm not sure, but I think it was being checked by warden.authenticated?(options[:scope]) on the trackable hook.

@feliperenan That makes sense. Is it possible to still do that? I'm not sure if we have access to the Warden instance or the resource's scope now, but I guess it would be safer to check whether the user is authenticated or not - e.g. in a speculative case where the `#sucess` method fails - but I might be overthinking 🤔 
Thoughts?

Reply:
Looking into the old code, I think we don't need check for `warden.authenticated?` since the user should be already authenticated in the trackable context. Maybe this check was there because we were just being too defensive. Also, I removed it from trackable in the master branch and all tests still pass (I think we can rely on Devise test suite 😅).

Reply:
Yeah, that makes sense. I guess we can continue with this implementation then. 👍